rpcd/playbooks/roles/elasticsearch/tasks/upgrade_pre_flight.yml

```diff
@@ -13,31 +13,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-- name: Install Pre Flight Checks
-  template:
-    src: upgrade-pre-flight-checks.py.j2
-    dest: /opt/upgrade-pre-flight-checks.py
-    owner: root
-    group: root
-    mode: 0755
-  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
-  tags:
-    - logging-upgrade
-    - elasticsearch-upgrade
-    - es-pre-flight
+- name: "Get ElasticSearch version"
+  command: "dpkg-query --show --showformat='{$Version}' elasticsearch"
+  register: elasticsearch_installed_version
 
-- name: Run Pre Flight Checks
-  shell: /opt/upgrade-pre-flight-checks.py
-  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
-  register: pre_flight_status
-  ignore_errors: true
-  tags:
-    - logging-upgrade
-    - elasticsearch-upgrade
-    - es-pre-flight
+- name: "Check for ElasticSearch version > 2.x"
+  set_fact:
+    proceed_with_upgrade: false
+    latest_version: true
+  register: proceed_with_upgrade
```

Comment:
`set_fact` and `register` do different things. `set_fact` operates on the host that's running the task. `register` is a global thing (accessible from everywhere). I'm not sure how `proceed_with_upgrade.ansible_facts.proceed_with_upgrade` works though (or how that would work at all).